catalog/CatalogTypedefs.hpp

```diff
@@ -36,6 +36,10 @@ typedef int attribute_id;
 typedef std::size_t partition_id;
 typedef int numa_node_id;
 
+// Catalog ids use negative values as invalid ids. Mark -1 as constant invalid
+// id for the catalog ids.
+constexpr int kInvalidCatalogId = -1;
```

Comment:
Move this below the comment `This depends on all the above id types being typedefed to int...`, as the comment applies to it as well.

Also, we need a separate PR to replacing all `-1` that represents invalid ids with this new value in `/catalog`.